thirdparty/openapi3/schemas/AuthorizationStateType.yaml

```diff
@@ -0,0 +1,7 @@
+title: AuthorizationStateType
+description: Transition state of thirdparty authorizations.
+type: string
+enum:
+  - WAITING_FOR_AUTHORIZATION_REQUEST
```

Comment:
https://github.com/mojaloop/thirdparty-scheme-adapter/blob/master/src/interface/api-outbound-template.yaml
Copied from the scheme-adapter. 
Seems to be isolated to the adapter? https://github.com/search?q=org%3Amojaloop+WAITING_FOR_AUTHORIZATION_REQUEST&type=code

Reply:
The Outbound responses are different from responses received from Switch. They differed because they contain details of state machine, also request will be different! they should be separated from the core Switch interfaces

Reply:
Stupid questions incoming. Why is important to communicate the details of the state machine to another service?

Reply:
If `currentState` is the only difference between outbound/switch interfaces, would using an optional `ExtensionList` be better for service specific stuff. Not sure but just tossing this out there.
@lewisdaly @eoln 
```
        extensionList:
          $ref: "#/components/schemas/ExtensionList"
          description: Optional extension, specific to deployment.
```

Lets try get this PR checked in before I work on the api doc generation tickets.

Reply:
other apis from sdk-scheme-adapters aren't using extension lists, so I followed this convention.